pgjdbc/src/test/java/org/postgresql/test/jdbc2/PreparedStatementTest.java

```diff
@@ -1767,4 +1769,96 @@ public void testNoParametersNPE() throws SQLException {
       // ignore
     }
   }
+
+  @Test
+  public void testBatchSelect() throws SQLException {
+    PreparedStatement pstmt = null;
+    PreparedStatement batchSelect = null;
+    ResultSet rs = null;
+    try {
+      pstmt = con.prepareStatement("INSERT INTO inttable (a) VALUES (?);"
+              + "INSERT INTO inttable (a) VALUES (?);"
+              + "INSERT INTO inttable (a) VALUES (?);"
+              + "INSERT INTO inttable (a) VALUES (?);");
+      ((PgStatement) pstmt).setPrepareThreshold(0);
+      for (int i = 0; i < 4; i++) {
+        pstmt.setInt(i + 1, i + 1);
+      }
+      pstmt.execute();
+      batchSelect =  con.prepareStatement("select * from inttable where a = ?;"
+          + "select * from inttable where a = ?;"
+          + "select * from inttable where a = ?;");
+      for (int i = 0; i < 3; i++) {
+        batchSelect.setInt(i + 1, i + 1);
+      }
+
+      List<Integer> results = new ArrayList<>();
+      boolean hasResults = batchSelect.execute();
+      while (hasResults) {
+        rs = batchSelect.getResultSet();
+        while (rs.next()) {
+          results.add(rs.getInt(1));
+        }
+        hasResults = batchSelect.getMoreResults();
+      }
+      assertEquals(Arrays.asList(1, 2, 3), results);
+    } catch (SQLException e) {
+      throw new RuntimeException(e);
+    } finally {
+      TestUtil.closeQuietly(rs);
+      TestUtil.closeQuietly(pstmt);
+      TestUtil.closeQuietly(batchSelect);
+    }
+  }
+
+  @Test
+  public void testBatchSelectWithPrepareThreshold1() throws SQLException {
```

Comment:
Just wondering: is it covered with existing `BinaryMode=ON` test parameter?
If not, we should probably consider adding `prepareThreshold` variation on a global level rather than adding individual tests.